tests/openvino/test_diffusion.py

```diff
@@ -438,6 +438,24 @@ def test_load_custom_weight_variant(self):
 
         np.testing.assert_allclose(ov_images, diffusers_images, atol=1e-4, rtol=1e-2)
 
+    @parameterized.expand(SUPPORTED_ARCHITECTURES)
+    @require_diffusers
+    def test_static_shape_image_generation(self, model_arch):
+        pipeline = self.OVMODEL_CLASS.from_pretrained(MODEL_NAMES[model_arch], compile=False)
+        pipeline.reshape(batch_size=1, height=64, width=32)
+        pipeline.compile()
+        # generation with incompatible size
+        height, width, batch_size = 64, 64, 1
+        inputs = self.generate_inputs(height=height, width=width, batch_size=batch_size)
+        inputs["output_type"] = "pil"
+        image = pipeline(**inputs).images[0]
+        self.assertTupleEqual(image.size, (32, 64))
+        # generation without height / width provided
+        inputs.pop("height")
+        inputs.pop("width")
+        image = pipeline(**inputs).images[0]
+        self.assertTupleEqual(image.size, (32, 64))
```

Comment:
please add an assertion for warning/not warning when shapes are mismatched/matched.

Reply:
Here are some examples from unittest docs https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarns
I believe you should use https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertLogs if the warning is done by the logger. This can be used to make sure the warning is there when we want it to be there and avoid false positives.